src/DynamoCore/Graph/Nodes/FunctionCallNodeController.cs

```diff
@@ -53,8 +54,19 @@ protected FunctionCallNodeController(T def)
         /// <param name="inputAstNodes">Arguments to the function call.</param>
         public IEnumerable<AssociativeNode> BuildAst(NodeModel model, List<AssociativeNode> inputAstNodes)
         {
+            return BuildAst(model, inputAstNodes, CompilationContext.DeltaExecution); 
+        }
+
+        /// <summary>
+        ///     Produces AST for a function call. Takes into account multi-outputs and partial application.
+        /// </summary>
+        /// <param name="model">NodeModel to produce an AST for.</param>
+        /// <param name="inputAstNodes">Arguments to the function call.</param>
+        /// <param name="context">Compilation context</param>
+        public IEnumerable<AssociativeNode> BuildAst(NodeModel model, List<AssociativeNode> inputAstNodes, CompilationContext context)
+        {
```

Comment:
Perhaps this should have been an `internal` rather than `public`? One less public API is always better if possible.